src/builder/modules/Generator.res

```diff
@@ -156,6 +226,53 @@ let drawFoldLine = (from: Builder.position, to: Builder.position) => {
     )
 }
 
+let drawFoldLinePath = (points: array<Builder.position>, ~close: bool=false, ()) => {
```

Comment:
Similar to my comment about the `drawLine()` I wonder if there is really much need for two similar fold line functions? This is marginally nicer to use for multiple joined lines, but it's still trivial to do the equivalent with `drawFoldLine()`.

Reply:
The path functions will be very useful for drawing fold outlines, since the marching squares algorithm for finding the outline will return an array of positions, and the drawPath() function will be able to take those as a parameter.

Reply:
For an outlining feature, that path drawing will need to occur with the Generator_Builder (it won't be possible for the Generator_Builder to call functions on the Generator module).

The pros and cons here are probably similar to the line drawing function chat above. I.e. potentially has benefit when there are many points, but could also be achieved easily with `drawFoldLine()` within a loop?

Reply:
One of the reasons for using drawPath() over drawLine() in a loop is that drawPath() renders once when it is called- it isn't known how much this could affect rendering times, but it should be a good idea to draw as few times as needed, especially if generators start drawing folds, tabs, and labels each during multiple steps.

Reply:
> it isn't known how much this could affect rendering times

As long as we're not drawing hundreds/thousands the there should be no noticeable difference (in the order of a few milliseconds) so it's not something to be concerned about for now.

Something else that I suspect will come up is that line _offset_ matter.

* With drawPath() we won't have control of the offset of each line segment.
* With drawLine() we will have control of the offset of each line (i.e. reversing the direction of the line drawing changes which side the offset is applied to).

There are some reasons what drawLine() will be favourable here, but let me know if you'd like me to explain further.